lib/plugins/aws/deploy/compile/events/s3/index.js

```diff
@@ -38,6 +39,43 @@ class AwsCompileS3Events {
               if (event.s3.event) {
                 notificationEvent = event.s3.event;
               }
+              if (event.s3.rules) {
+                if (!event.s3.event) {
```

Comment:
hmmm why are we throwing an error if the user defined rules, but didn't define events? What should happen in that case is that the default ObjectCreated event will be added to CF just like I mentioned in the previous comment...or maybe I'm missing something obvious here?